iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ShuntXml.java

```diff
@@ -51,8 +51,11 @@ protected void writeRootElementAttributes(ShuntCompensator sc, VoltageLevel vl,
             IidmXmlUtil.assertMinimumVersion(getRootElementName(), SHUNT_NON_LINEAR_MODEL, IidmXmlUtil.ErrorMessage.NOT_SUPPORTED, IidmXmlVersion.V_1_3, context);
         }
         IidmXmlUtil.runUntilMaximumVersion(IidmXmlVersion.V_1_2, context, () -> {
-            XmlUtil.writeDouble(B_PER_SECTION, sc.getModel(ShuntCompensatorLinearModel.class).getBPerSection(), context.getWriter());
-            context.getWriter().writeAttribute(MAXIMUM_SECTION_COUNT, Integer.toString(sc.getMaximumSectionCount()));
+            ShuntCompensatorModel model = sc.getModel();
+            double bPerSection = model instanceof ShuntCompensatorLinearModel ? ((ShuntCompensatorLinearModel) model).getBPerSection() : sc.getB();
+            XmlUtil.writeDouble(B_PER_SECTION, bPerSection, context.getWriter());
+            int maximumSectionCount = model instanceof ShuntCompensatorLinearModel ? sc.getMaximumSectionCount() : 1;
+            context.getWriter().writeAttribute(MAXIMUM_SECTION_COUNT, Integer.toString(maximumSectionCount));
             context.getWriter().writeAttribute("currentSectionCount", Integer.toString(sc.getSectionCount()));
```

Comment:
I'm not totally sure this is OK. The new shunt API wasn't available before XIIDM v1.3. If the model is non-linear, should we throw an exception instead? From my point of view, it's not equivalent and should not be supported